streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java

```diff
@@ -1038,13 +1038,15 @@ public void shouldThrowConfigExceptionWhenOptimizationConfigNotValueInRange() {
         assertThrows(ConfigException.class, () -> new StreamsConfig(props));
     }
 
+    @SuppressWarnings("deprecation")
     @Test
     public void shouldSpecifyRocksdbWhenNotExplicitlyAddedToConfigs() {
         final String expectedDefaultStoreType = StreamsConfig.ROCKS_DB;
         final String actualDefaultStoreType = streamsConfig.getString(DEFAULT_DSL_STORE_CONFIG);
         assertEquals("default.dsl.store should be \"rocksDB\"", expectedDefaultStoreType, actualDefaultStoreType);
     }
 
+    @SuppressWarnings("deprecation")
     @Test
     public void shouldSpecifyInMemoryWhenExplicitlyAddedToConfigs() {
```

Comment:
Can you add some basic tests for the new store config? Basically something like what we have for the old configs --  ie the default is rocksdb and that the built-in InMemoryDslStoreSuppliers -- but instead of the bad-enum one, just make sure you can actually pass in a custom implementation

Reply:
done